### ISSUE TITLE
Rec model server crashes on vector features with missing geometry

### ISSUE THREAD

**Issue by davemfish** [natcap/invest#304]
Here's the tail of the server's log:
```

09/07/2020 14:32:25  natcap.invest.recreation.recmodel_server INFO     updating quadtree shape at ./recserver_cache_py36/ef4cbfcf-caf9-4085-9516-e2d363ec880a/local_qt/local_qt.shp
09/07/2020 14:32:28  natcap.invest.recreation.recmodel_server INFO     Creating a copy of the input shapefile
09/07/2020 14:32:28  natcap.invest.recreation.recmodel_server INFO     in a _calc_poly_process, loading ./recserver_cache_py36/ef4cbfcf-caf9-4085-9516-e2d363ec880a/local_qt/local_qt.pickle
09/07/2020 14:32:28  natcap.invest.recreation.recmodel_server INFO     in a _calc_poly_process, loading ./recserver_cache_py36/ef4cbfcf-caf9-4085-9516-e2d363ec880a/local_qt/local_qt.pickle
09/07/2020 14:32:29  natcap.invest.recreation.recmodel_server INFO     local qt load took 0.87s
09/07/2020 14:32:29  natcap.invest.recreation.recmodel_server INFO     local qt load took 0.88s
09/07/2020 14:33:06  natcap.invest.recreation.recmodel_server INFO     testing polygons against quadtree
Process Process-2213:
Process Process-2214:
Traceback (most recent call last):
Traceback (most recent call last):
  File "/usr/local/recreation-server/invest-davemfish/invest-env-py36/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/local/recreation-server/invest-davemfish/invest-env-py36/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/recreation-server/invest-davemfish/invest-env-py36/lib/python3.6/site-packages/natcap.invest-3.7.0.post141+hc694373c56a9.d20190708-py3.6-linux-x86_64.egg/natcap/invest/recreation/recmodel_server.py", line 713, in _calc_poly_pud
    poly_geom = poly_feat.GetGeometryRef()
AttributeError: 'NoneType' object has no attribute 'GetGeometryRef'
  File "/usr/local/recreation-server/invest-davemfish/invest-env-py36/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/local/recreation-server/invest-davemfish/invest-env-py36/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/recreation-server/invest-davemfish/invest-env-py36/lib/python3.6/site-packages/natcap.invest-3.7.0.post141+hc694373c56a9.d20190708-py3.6-linux-x86_64.egg/natcap/invest/recreation/recmodel_server.py", line 713, in _calc_poly_pud
    poly_geom = poly_feat.GetGeometryRef()
AttributeError: 'NoneType' object has no attribute 'GetGeometryRef'
```

It's processing AOI features here. It may make sense to "clean up" the AOI in `recmodel_client.py` before sending it to the server because both the client and server use the AOI features to aggregate other data and it's important they work with the same set of AOI features. In general it seems okay to skip/ignore features with missing geometry, rather than crashing.

**Comment by davemfish** [comment]
If this is difficult to reproduce with synthetic data, I have the AOI cached in my home dir on the rec server as `aoi_for_issue304.zip`

**Comment by davemfish** [comment]
And it also crashes on vector's with no layers.

```
Traceback (most recent call last):
  File "/usr/local/recreation-server/invest-davemfish/invest-env-py36/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/local/recreation-server/invest-davemfish/invest-env-py36/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/recreation-server/invest-davemfish/invest-env-py36/lib/python3.6/site-packages/natcap.invest-3.7.0.post141+hc694373c56a9.d20190708-py3.6-linux-x86_64.egg/natcap/invest/recreation/recmodel_server.py", line 709, in _calc_poly_pud
    aoi_layer = aoi_vector.GetLayer()
AttributeError: 'NoneType' object has no attribute 'GetLayer'
```

**Comment by davemfish** [comment]
@dcdenu4 On closer examination, it may have just been a coincidence that these errors were in the server log while the server appeared unresponsive. They show that a process crashed, but the server is still online and accepting connections.

However, when a process does crash with one of these exceptions, it never makes it back to the client. So the client is stuck on `Contacting server, please wait...`

And if the server ever truly was unresponsive (as it seemed to be when running our invest CI) I'm not sure why that is.

**Comment by davemfish** [comment]
fixed in #315
